cmd/osm-controller/osm-controller.go

```diff
@@ -215,7 +215,7 @@ func main() {
 		cfg,
 		endpointsProviders...)
 
-	proxyRegistry := registry.NewProxyRegistry()
+	proxyRegistry := registry.NewProxyRegistry(&registry.KubeProxyServiceMapper{KubeController: kubernetesClient})
```

Comment:
This seems like a nice abstraction.  You have the ProxyRegistry, and then you have this other facility, which knows how to get the services for a given proxy.